modules/biobambam/bamsormadup/main.nf

```diff
@@ -0,0 +1,38 @@
+process BIOBAMBAM_BAMSORMADUP {
+    tag "$meta.id"
+    label 'process_medium'
+
+    conda (params.enable_conda ? "bioconda::biobambam=2.0.183" : null)
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/biobambam:2.0.183--h9f5acd7_1':
+        'quay.io/biocontainers/biobambam:2.0.183--h9f5acd7_1' }"
+
+    input:
+    tuple val(meta), path(bam)
+
+    output:
+    tuple val(meta), path("*.bam")        , emit: bam
+    tuple val(meta), path("*.metrics.txt"), emit: metrics
+    path "versions.yml"                   , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+    """
+    bamsormadup \\
+        $args \\
+        I=$bam \\
+        O=${prefix}.bam \\
+        M=${prefix}.metrics.txt \\
```

Comment:
Please remove `metrics`, file naming should be up to the pipeline developer [^1]

[^1]: https://nf-co.re/developers/modules#naming-conventions point 5

Reply:
how would you suggest differentiating the filename? Or do you mean remove `M=xxx` and moving it to `$args`?

Reply:
Just I mean just specify as `{prefix}.txt`, unless the tool can also produce other `.txt` files? Otherwise `.txt` should be enough to differentiate - and a pipeline developer can add `metrics` themselves if they want.

Reply:
Allright, thanks. This is resolved than.

Reply:
I see you reverted back to `metrics`, is that required by the tool in the end?